### ISSUE TITLE
The result of training always generates all-0 value maps

### ISSUE THREAD

**Issue by SII-Ferenas** [halbielee/SEC_pytorch#3]
I don't know why the model after being trained in train.script always generates all-0 value maps, but I test the best model you provide shows the result is right...

**Comment by halbielee** [comment]
Hi!

I read your question, but I don't know what is the exact reason for those all 0 values. 
If there is not error message for training, please check the loss. It should decrease! If the loss has value 0.69xx, it seems to be fail to train. 

This model is also based on the ImageNet Pretraining model!

If you have any question, please leave another message!

**Comment by SII-Ferenas** [comment]
Thanks for your reply, I set the right folder for the training and use the vgg_20M you provide, and the sum of 3 losses always converges  to about 7, and the test result generates all-0 maps, which means that the model recognizes all pixels in map as the backgroud, I really don't know what's going on... btw, the preprossing of image in traing and test process is not the same...

**Comment by halbielee** [comment]
Hmm, the sum of three losses is too high, I think the training is not properly working. Please check the whole pretrained weight is loaded to model well. 

Please check the closed issue about the number of loss https://github.com/halbielee/SEC_pytorch/issues/1

**Comment by SII-Ferenas** [comment]
Thanks for your reply. it seems that the question he met is nearly the same as mine , but I used the vgg-16 pretrained model you provide to proceed the whole training process. So the problem might be from the vgg16_20M files you provide ?

**Comment by halbielee** [comment]
I think the vgg16_20M file is the exact ImageNet pretrained file for this work, but you might be right. You can try to convert the trained model from caffe or you could use another converted pretrained model.

There are several pretrained model files! 

This might be another option! [https://github.com/visinf/1-stage-wseg](https://github.com/visinf/1-stage-wseg)

From the page, you can download the converted one.

**Comment by SII-Ferenas** [comment]
![image](https://user-images.githubusercontent.com/55052174/87546381-0e1b5b00-c6dc-11ea-8148-c79b8d0d7488.png)
Thanks for your reply, but I have tried the pretrained vgg_20M in this links, the results are nearly the same.. I am really curious about the problems, since the test.py and evaluation.py you provide is exactly right, I did not modify any hyperparameters except for some folders.  Would you mind leaving me a more convenient contact approach ? I can talk directly to you if not bothering you. Here is my email-address ferenas@sjtu.edu.cn.

**Comment by halbielee** [comment]
Hello Fei!

I upload the pretrained model of PascalVOC again!!
I checked it works.
I hope this is helpful!

Thank you!

**Comment by SII-Ferenas** [comment]
ok，do you find the problem？actually I am curious about the problems hah发自我的iPhone------------------ Original ------------------From: halbielee  <notifications@github.com>Date: Sun,Jul 19,2020 11:59 AMTo: halbielee/SEC_pytorch <SEC_pytorch@noreply.github.com>Cc: Ferenas <ferenas@sjtu.edu.cn>, Author <author@noreply.github.com>Subject: Re: [halbielee/SEC_pytorch] The result of training always generatesall-0 value maps (#3)
Hello Fei!
I upload the pretrained model of PascalVOC again!!
I checked it works.
I hope this is helpful!
Thank you!

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/halbielee/SEC_pytorch/issues/3#issuecomment-660582545",
"url": "https://github.com/halbielee/SEC_pytorch/issues/3#issuecomment-660582545",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by halbielee** [comment]
Hmm There might be a difference in weight value in DeepLab official caffemodel and SEC official caffemodel. 

If we use VGG16_20M.caffemodel from DeepLab official code, SEC both in caffe, pytorch) can not learn properly.

It is hard to distinguish the exact difference. :(

I hope you can train this model well!

**Comment by SII-Ferenas** [comment]
So now you use the pretrained model from Deeplab official code? haha, it is actually a funny problem

**Comment by halbielee** [comment]
@Ferenas-ZhangFei 

I changed README.md file and uploaded pretrained model is from SEC official code.
The previous pretrained model is from DeepLab official code.

So, you should use prerained model from SEC to reproduce and the uploaded version is from SEC

Here is the [link](https://drive.google.com/file/d/1tAnc1fDttigaer1UC5rypPGTZUt2GGeK/view)

**Comment by SII-Ferenas** [comment]
Okay , I use the pretrained model you provide and it actually works!!!! Thanks a lot, having contacted you in Skype
